### PR TITLE
Optimize min/max aggregation with BlockPositionState

### PR DESCRIPTION
**Pull request by wenleix**
min/max have the same issue with arbitrary when it's aggregating over varchar/structural types (#9696). While min/max over array/map is rare, we see increasing usage case for min/max over varchar/row, which causes GC pressure.

### CODE REVIEW

File: presto-product-tests/src/main/resources/sql-tests/testcases/math_functions/checkMathFunctionsRegistered.result [hunk 1]
```diff
@@ -49,9 +49,9 @@
  log | double | double, double | scalar | true | logarithm to given base |
  log10 | double | double | scalar | true | logarithm to base 10 |
  log2 | double | double | scalar | true | logarithm to base 2 |
- max | T | T | aggregate | true | Returns the maximum value of the argument |
```
Comment (highker) [inline presto-product-tests/src/main/resources/sql-tests/testcases/math_functions/checkMathFunctionsRegistered.result]: why is this necessary?

File: presto-main/src/main/java/com/facebook/presto/operator/aggregation/AbstractMinMaxAggregationFunction.java [hunk 2]
```diff
@@ -0,0 +1,338 @@
...
+        else if (type.getJavaType() == boolean.class) {
+            stateInterface = NullableBooleanState.class;
+            stateSerializer = StateCompiler.generateStateSerializer(stateInterface, classLoader);
+            inputFunction = BOOLEAN_INPUT_FUNCTION;
+            combineFunction = BOOLEAN_COMBINE_FUNCTION;
+            outputFunction = BOOLEAN_OUTPUT_FUNCTION;
+        }
+        else {
+            stateInterface = BlockState.class;
+            stateSerializer = new BlockStateSerializer(type);
```
Comment (highker) [inline presto-main/src/main/java/com/facebook/presto/operator/aggregation/AbstractMinMaxAggregationFunction.java]: I think it would be much cleaner to squash these two commits; I was very confused when I first saw this given this is not an optimized serializer.

**Comment by wenleix** [comment]
The first commits partially reverts the MIN/MAX related part of https://github.com/prestodb/presto/commit/3030ab66d4c9610ec85efdc762180c663e953cde, which is part of PR https://github.com/prestodb/presto/pull/6627

File: presto-main/src/main/java/com/facebook/presto/operator/aggregation/AbstractMinMaxAggregationFunction.java [hunk 3]
```diff
@@ -0,0 +1,325 @@
...
+        catch (Throwable t) {
+            throw internalError(t);
+        }
+    }
+
+    public static void minCombine(Type type, BlockPositionState state, BlockPositionState otherState)
+    {
+        if (state.getBlock() == null) {
+            state.setBlock(otherState.getBlock());
+            state.setPosition(otherState.getPosition());
```
[the remark this answers is not in the document]
Comment (wenleix) [reply presto-main/src/main/java/com/facebook/presto/operator/aggregation/AbstractMinMaxAggregationFunction.java]: @highker : But it's just two lines and seems quite straightforward. Not sure if worthy it :)

File: presto-main/src/main/java/com/facebook/presto/operator/aggregation/AbstractMinMaxAggregationFunction.java [hunk 4]
```diff
@@ -0,0 +1,325 @@
...
+        }
+        else {
+            return ImmutableList.of(
+                    new ParameterMetadata(STATE),
+                    new ParameterMetadata(BLOCK_INPUT_CHANNEL, type),
+                    new ParameterMetadata(BLOCK_INDEX));
+        }
+    }
+
+    public static void input(MethodHandle methodHandle, NullableDoubleState state, double value)
```
[the remark this answers is not in the document]
Comment (wenleix) [reply presto-main/src/main/java/com/facebook/presto/operator/aggregation/AbstractMinMaxAggregationFunction.java]: Good idea!

File: presto-main/src/main/java/com/facebook/presto/operator/aggregation/AbstractMinMaxAggregationFunction.java (hunk 3 above)
[the remark this answers is not in the document]
Comment (wenleix) [reply presto-main/src/main/java/com/facebook/presto/operator/aggregation/AbstractMinMaxAggregationFunction.java]: @highker  : I think this might be better: to have `BlockPostionState` manually written, and has a single method called `setBlockAndPosition`.

It doesn't make sense to set them separately.

**Comment by wenleix** [comment]
Part of fix to #9553
